### ISSUE TITLE
llm_tricks里面的dpo_examples里面训练单轮，迭代一次后，policy_model的logits都变成nan了

### ISSUE THREAD

**Issue by oppppp** [mst272/LLM-Dojo#23]
你好，我想用dpo算法训练bilibili的模型IndexTeam/Index-1___9B-Chat，用来对给出的一篇文章内容生成合适的标题。
训练数据：
{"prompt":"文章正文。请生成一个与给出的内容十分匹配的标题："，"chosen":"评估匹配的标题"，"rejected": "评估不匹配的标题"}。
机器配置：
v100 32g
参数设置：
batch_size=1，lr=1e-5, 其他保持与原始代码不变
policy_model和reference_model，都因为显存不够，设置成 f16和load_in_4bit。

问题1:
第一步，policy_model的logits正常，得到loss0.69, 进行loss.backward()后，后面policy_model的logits就全部变为nan了。
解决方案1:
使用 with torch.autograd.detect_anomaly():
           loss.backward()
监控，显示为RuntimeError: Function 'LogSigmoidBackward0' returned nan values in its 0th output. 

有人知道为啥会出现这种情况吗？

**Comment by mst272** [comment]
数据格式不太对吧，应该是带user和assistant的，可以看一下data\dpo_multi_data.jsonl

**Comment by mst272** [comment]
这个问题是由于f16精度导致的，改为fp32是可以的。你可以换个更小的模型试一下

**Comment by oppppp** [comment]
嗯嗯，确实是精度导致的
